### DISCUSSION TITLE
Text to speech: Streaming vs REST Rate limit

### DISCUSSION THREAD

**Discussion by nitin-airindia** [deepgram/community#960] [General help] [answered]
I am currently evaluating the use of Deepgram's text-to-speech capabilities for my backend application, which involves stt/llm/tts functionalities.
I noticed that the WebSocket API allows for up to 40 concurrent connections, while the Batch API supports 2 concurrent requests with a limit of 480 requests per minute. So. does this means WebSocket API can handle 20 times more text to speech conversion request/min (i.e. 9600 coversions/min)? I would like to understand which method would provide me with more requests per minute for my application, especially when handling multiple conversations simultaneously.

**Comment by jkroll-deepgram** [comment, marked as answer]
Hi @nitin-airindia, the websocket API allows for more flexible concurrency, and is a good choice for higher-volume applications.

While the batch API supports 2 concurrent requests, the average TTS REST API call is only 250 milliseconds, which can cover tens of seconds of a conversation, so in practice two concurrent requests can support dozens or hundreds of concurrent conversations, depending on how often the TTS speaks in the conversation.
